src/ui/components/Library/Team/View/Tests/Overview/TestDetails.tsx

```diff
@@ -1,13 +1,93 @@
-import { TestExecution } from "shared/test-suites/TestRun";
+import orderBy from "lodash/orderBy";
+
+import { Recording } from "shared/graphql/types";
+import { GroupedTestRun } from "shared/test-suites/TestRun";
 import { testFailed, testPassed } from "ui/utils/testRuns";
 
-import { ReplayList } from "./ReplayList";
+import { TestSuitePanelMessage } from "../../TestSuitePanelMessage";
+import { Execution } from "./Execution";
+import { StatusIcon } from "./StatusIcon";
+
+export function TestDetails({ testRuns }: { testRuns: GroupedTestRun[] }) {
+  const sortedTestRuns = orderBy(testRuns, "date", "desc");
+
+  if (!sortedTestRuns.length) {
+    return (
+      <div className="flex flex-col gap-1 p-2">
+        <TestSuitePanelMessage>No test runs found</TestSuitePanelMessage>;
+      </div>
+    );
+  }
+
+  return (
+    <div className="flex flex-col gap-1 p-2">
+      {sortedTestRuns.map((t: any) => (
+        <TestRun testRun={t} key={t.testRunId} />
+      ))}
+    </div>
+  );
+}
+
+function TestRun({ testRun }: { testRun: GroupedTestRun }) {
+  const executionsWithReplays = testRun.executions.filter(
+    e => e.recordings.length && (testPassed(e) || testFailed(e))
+  );
+
+  if (!executionsWithReplays.length) {
+    return null;
+  }
+
+  const sortedExecutions = orderBy(executionsWithReplays, "createdAt", "desc");
+
+  return (
+    <div className="flex flex-col rounded-md bg-chrome">
+      {sortedExecutions.map((e, i) => (
+        <Execution execution={e} key={i} />
+      ))}
+    </div>
+  );
+}
+
+export function getReplayResult(result: string, index: number, total: number) {
```

Comment:
@ryanjduffy This makes a _lot_ of assumptions, so feel free to flag anything I'm missing here.